### PR TITLE
CSVS-27 ZooKeeper service and tests were refactored

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

File: src/test/scala/com/bwsw/cloudstack/vault/server/vault/VaultServiceTestSuite.scala [hunk 1]
```diff
@@ -37,15 +37,15 @@ class VaultServiceTestSuite extends FlatSpec with TestData with BaseTestSuite {
     assert(expectedToken == token)
   }
 
-  "revokeToken" should "revoke token and return path to secret data" in {
+  "revokeToken" should "revoke token" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/vault/server/vault/VaultServiceTestSuite.scala]: "revokeToken" should "revoke a token"

File: src/main/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperService.scala [hunk 2]
```diff
@@ -18,78 +18,96 @@
...
-    * Launches task for getting data from zNode in ZooKeeper server.
-    * Will restarts task if zookeeper server is unavailable.
+    * Gets data from zNode in ZooKeeper server.
+    * Will be waiting if zookeeper server is unavailable.
     *
     * @param path String with path to zNode data
     *
-    * @return String with data stored in the zNode
-    * @throws ZooKeeperCriticalException if zNode already does not exist.
+    * @return String with data stored in the zNode if zNode exist
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperService.scala]: Data as string stored in zNode if zNode exist

File: src/test/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperServiceTestSuite.scala [hunk 3]
```diff
@@ -2,133 +2,72 @@ package com.bwsw.cloudstack.vault.server.zookeeper
...
-        throw new ZooKeeperCriticalException(new Exception("test exception"))
-      }
-    }
-
-    val zooKeeperService = new ZooKeeperService(zooKeeperTaskCreator, zooKeeperServiceSettings)
-
-    assertThrows[ZooKeeperCriticalException] {
-      zooKeeperService.deleteNode(expectedPath)
-    }
+  "getDataIfNodeExist" should "return None if node does not exist" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperServiceTestSuite.scala]: replace with getNodeData

File: src/test/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperServiceTestSuite.scala [hunk 4]
```diff
@@ -2,133 +2,72 @@ package com.bwsw.cloudstack.vault.server.zookeeper
...
-
-    val zooKeeperTaskCreator = new ZooKeeperTaskCreator(zooKeeperTaskCreatorSettings) {
-      override def createCheckExistNodeTask(path: String)(): Boolean = {
-        assert(expectedPath == path, "path is wrong")
-        throw new ZooKeeperCriticalException(new Exception("test exception"))
-      }
-    }
-
-    val zooKeeperService = new ZooKeeperService(zooKeeperTaskCreator, zooKeeperServiceSettings)
+  "isExistNode" should "return false if node does not exist" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperServiceTestSuite.scala]: replace with doesNodeExist

File: src/main/scala/com/bwsw/cloudstack/vault/server/vault/VaultService.scala [hunk 5]
```diff
@@ -73,10 +74,10 @@ class VaultService(vaultRest: VaultRestRequestCreator,
     *
     * @param tokenId UUID of token for revoke
     *
-    * @return String of path to secret of revoked token
+    * @return String list of policies name of revoked token
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/vault/server/vault/VaultService.scala]: List of names of revoked token policies

File: src/test/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultControllerTestSuite.scala [hunk 6]
```diff
@@ -690,6 +693,24 @@ class CloudStackVaultControllerTestSuite extends FlatSpec with BaseTestSuite wit
     assert(revokedTokenList == checkedRevokedTokens)
   }
 
+  "createMissingAccountTokenTags" should "trow IllegalArgumentException if tag key does not in (VaultRO, VaultRW)" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultControllerTestSuite.scala]: "createMissingAccountTokenTags" should "throw IllegalArgumentException if tag key is not one of (VaultRO, VaultRW)"

File: src/test/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultControllerTestSuite.scala [hunk 7]
```diff
@@ -772,9 +793,47 @@ class CloudStackVaultControllerTestSuite extends FlatSpec with BaseTestSuite wit
     (vaultService, zooKeeperService)
   }
 
-  private def getServicesForTestDeletionEventIfZNodeExist(tokenReadPath: String,
-                                                          tokenWritePath: String,
-                                                          testSecretPathFromToken: String)
+  private def getServicesForTestEntityCreationIfTokensDoesNotExist(expectedReadPolicy: Policy,
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultControllerTestSuite.scala]: getServicesForTestEntityCreationIfTokensDoNotExist

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/test/scala/com/bwsw/cloudstack/vault/server/vault/VaultServiceTestSuite.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/vault/server/vault/VaultServiceTestSuite.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperService.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperService.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperServiceTestSuite.scala (hunk 3 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperServiceTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperServiceTestSuite.scala (hunk 4 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/vault/server/zookeeper/ZooKeeperServiceTestSuite.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/vault/server/vault/VaultService.scala (hunk 5 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/vault/server/vault/VaultService.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultControllerTestSuite.scala (hunk 6 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultControllerTestSuite.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultControllerTestSuite.scala (hunk 7 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultControllerTestSuite.scala]: ok

**Review by placccebo — APPROVED** [review]

**Pull request merged**